src/Microsoft.TemplateEngine.Edge/Settings/SettingsLoader.cs

```diff
@@ -268,6 +281,15 @@ private IEnumerable<MountPointInfo> FindMountPointsToScan(bool forceRebuild)
                     // TODO: This should never happen - throw an error?
                     continue;
                 }
+
+                //try to demand mount point: if the mount point is not available, the method returns false
+                //if the mount point is not available, we skip it so it doesn't cause exception when scanning it
+                if (!_mountPointManager.TryDemandMountPoint(mountPoint, out IMountPoint mp))
+                {
+                    continue;
+                }
+                _mountPointManager.ReleaseMountPoint(mp);
```

Comment:
What are the semantics of `ReleaseMountPoint`? I see `TryDemandMountPoint` calls throughout the codebase that seemingly leak the `IMountPoint`, i.e. they don't call `ReleaseMountPoint` on it afterwards, which leads me to believe that it's not just a dispose-like call. Does it really have no unintended side-effects? What would happen if the call was omitted? Thanks!

Reply:
I understand that `Release` is required dispose for `Demand`.

We have 2 implementation of mount point:
- [FileSystemMountPointFactory](https://github.com/dotnet/templating/blob/fad2cdb935ff5042e403963ad4f18d46b5e353f7/src/Microsoft.TemplateEngine.Edge/Mount/FileSystem/FileSystemMountPointFactory.cs#L40) - disposes nothing
- [ZipFileMountPointFactory] (https://github.com/dotnet/templating/blob/fad2cdb935ff5042e403963ad4f18d46b5e353f7/src/Microsoft.TemplateEngine.Edge/Mount/Archive/ZipFileMountPointFactory.cs#L84) - release is disposing previously created [ZipArchive](https://docs.microsoft.com/en-us/dotnet/api/system.io.compression.ziparchive?view=net-5.0)

You are correct that it's not used everywhere, however seems to be needed. It definitely does no harm, without it zip archive mount points are left undisposed.